public/content/documentation/web/component/password-input.md

```diff
@@ -0,0 +1,218 @@
+
+## Gherkin
+
+### #a11y - Web Accessibility Acceptance Criteria
+
+How to test a header
+
+GIVEN THAT I am on a page with a header landmark
+
+1. Keyboard for mobile & desktop
+
+   - WHEN I use the tab key to move focus to the password input I SEE focus is strongly visually indicated
+   - THEN when I use the tab key to move focus to the show/hide password feature I SEE its name, role and state
+   - THEN when I use the show/hide password feature I SEE the state of the password visibility (with or without characters entered)
+
+2. Desktop screenreader
+   - WHEN I use a desktop screenreader (NVDA, JAWS, VoiceOver) AND I use the tab key to move focus to the password input
+     - I HEAR Its purpose is clear
+     - I HEAR It identifies itself as a text input
+     - I HEAR Hints or errors are read after the label (Ex: Password formatting)
+     - I HEAR It expresses if the password is being shown and if applicable: required, disabled / dimmed / unavailable
+   - THEN when I use the tab key to move focus to the show/hide password feature I HEAR its name, role and state
+   - THEN when I use the show/hide password feature I HEAR the state of the password visibility (with or without characters entered)
+
+3. Mobile screenreader
+   - WHEN I use a mobile screenreader (Talkback, VoiceOver) AND I swipe to focus on a password input
+     - I HEAR Its purpose is clear
+     - I HEAR It identifies itself as a text input
+     - I HEAR Hints or errors are read after the label (Ex: Password formatting)
+     - I HEAR It expresses if the password is being shown and if applicable: required, disabled / dimmed / unavailable
+
+
+Full information: [https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/component/password-input](https://www.magentaa11y.com/MagentaA11yV2#/web-criteria/component/password-input)
+
+## Code Examples
+
+### Checkbox variant
+   - This semantic HTML contains all accessibility features by default.
+   - Placing the show password checkbox ahead of the password input increases discoverability for screen reader users.
+     - CSS pseudo elements are used in the checkbox label to express its state on focus of the password input.
+
+```html
+<form class="password-container">
+  <label for="password">
+    Password
+    <span class="helper" aria-hidden="true">
+      Required
+    </span>
+  </label>
+  <div id="password-toggle" class="password-toggle">
+    <input type="checkbox" id="show-password">
+    <label for="show-password">
+      Show Password
+    </label>
+  </div>
+  <input type="password"
+         id="password"
+         aria-describedby="password-toggle password-hint"
+         aria-required="true"
+         required>
+  <div id="password-hint" class="hint">
+    Use any length of characters including emojis.
+  </div>
+</form>
+```
+
+<example>
```

Comment:
Let's add a TODO for the example; at this time the "show password" checkbox doesn't show/hide the text entered.